provider/scripts/k8s/offline_store_pandas_runner.py

```diff
@@ -159,10 +164,10 @@ def download_blobs_to_local(container_client, blob, local_filename):
     
     print(f"downloading {blob} to {local_filename}")
     if not os.path.isdir(LOCAL_DATA_PATH):
-        os.makedirs(LOCAL_DATA_PATH)
+        os.makedirs(LOCAL_DATA_PATH, exist_ok=True)
 
     full_path = f"{LOCAL_DATA_PATH}/{local_filename}"
-    if blob[-4:] == ".csv" or blob[-8:] == ".parquet":
+    if blob[-4:] == ".csv" or blob[-8:] == ".parquet" or blob[-4:] == ".pkl":

```

Comment:
might be easier to split on the final "." and compare that to a set of valid extensions